selfservice/flow/recovery/handler.go

```diff
@@ -269,7 +269,7 @@ func (h *Handler) fetch(w http.ResponseWriter, r *http.Request, _ httprouter.Par
 	// Browser flows must include the CSRF token
 	//
 	// Resolves: https://github.com/ory/kratos/issues/1282
-	if f.Type == flow.TypeBrowser && !nosurf.VerifyToken(h.d.GenerateCSRFToken(r), f.CSRFToken) {
+	if f.ShouldEnforceCSRF() && !nosurf.VerifyToken(h.d.GenerateCSRFToken(r), f.CSRFToken) {
```

Comment:
This assertion has been weakened by `f.CSRFToken != ""`, please revert

Reply:
Have you updated the assertion @jonas-jonas?

Reply:
See https://github.com/ory/kratos/pull/2645#discussion_r969242435

We need to figure out a way to handle recovery initiated by admins here, so no. :)